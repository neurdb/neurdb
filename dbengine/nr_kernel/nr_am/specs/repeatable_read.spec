setup
{
  SET client_min_messages TO WARNING;
  DROP TABLE IF EXISTS rc_test;
  DROP EXTENSION IF EXISTS nram CASCADE;
  DROP ACCESS METHOD IF EXISTS nram;
  DROP FUNCTION IF EXISTS nram_tableam_handler(internal);
  CREATE EXTENSION nram;
  CREATE TABLE accounts (id INT PRIMARY KEY, balance INT) USING nram;
  INSERT INTO accounts VALUES (1, 100);
}

teardown
{
  DROP TABLE accounts;
}

session s1
setup
{
  BEGIN ISOLATION LEVEL READ COMMITTED;
}
step s1_read1  { SELECT balance FROM accounts WHERE id = 1; }
step s1_read2  { SELECT balance FROM accounts WHERE id = 1; }
step s1_commit { COMMIT; }

session s2
setup
{
  BEGIN;
}
step s2_update { UPDATE accounts SET balance = 50 WHERE id = 1; }
step s2_commit { COMMIT; }

permutation s1_read1 s2_update s2_commit s1_read2 s1_commit
