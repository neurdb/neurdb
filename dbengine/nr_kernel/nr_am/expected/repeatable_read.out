Parsed test spec with 2 sessions

starting permutation: s1_read1 s2_update s2_commit s1_read2 s1_commit
step s1_read1: SELECT balance FROM accounts WHERE id = 1;
balance
-------
    100
(1 row)

step s2_update: UPDATE accounts SET balance = 50 WHERE id = 1;
step s2_commit: COMMIT;
step s1_read2: SELECT balance FROM accounts WHERE id = 1;
balance
-------
    100
(1 row)

step s1_commit: COMMIT;
